client/src/main/java/io/pravega/client/netty/impl/SessionHandler.java

```diff
@@ -183,8 +188,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         getReplyProcessor(cmd).ifPresent(processor -> {
             try {
                 processor.process(cmd);
-            } catch (Exception e) {
-                processor.processingFailure(e);
+            } catch (Throwable e) {
+                processor.processingFailure(new ExecutionException(e));
```

Comment:
Why do you wrap it in this? Can we change `processingFailure` to accept a `Throwable` instead?

Reply:
Yes, we could change `io.pravega.shared.protocol.netty.ReplyProcessor#processingFailure(Exception error)` to `io.pravega.shared.protocol.netty.ReplyProcessor#processingFailure(Throwable error)` . Tracking this with issue #3596